resdk/resolwe.py

```diff
@@ -155,29 +147,29 @@ def endswith_colon(schema, field):
             process['persistence'] = persistence_map[process['persistence']]
 
         try:
-            server_process = self.api.process.get(slug=process['slug'], ordering='-version', limit=1)
+            server_process = self.process.filter(slug=process['slug'], ordering='-version', limit=1)
 
             if len(server_process) == 1:
                 server_process = server_process[0]
-
-                if process['version'] > server_process['version']:
-                    response = self.api.process.post(process)
-                else:
+                if not process['version'] > server_process['version']:
                     process['version'] = server_process['version'] + 1
-                    self.logger.warning("Process '%s' version increased automatically: %s",
-                                        slug,
-                                        self._version_int_to_string(process['version']))
-                    response = self.api.process.post(process)
+                    self.logger.warning(
+                        "Process '%s' version increased automatically: %s",
+                        slug,
+                        self._version_int_to_string(process['version']))
+
+                response = self.api.process.post(process)
 
             elif len(server_process) == 0:
                 response = self.api.process.post(process)
             else:
                 raise ValueError("Unexpected behaviour at get process with slug {}".format(slug))
 
+        # How to reproduce this? What scenario causes this exception?
```

Comment:
Updating processes is supported only on development servers, so this error will be raised on production server.